lib/crawly/pipelines/experimental/send_to_ui.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Crawly.Pipelines.Experimental.SendToUI do
+  @moduledoc """
+  """
+  @behaviour Crawly.Pipeline
+
+  require Logger
+
+  @impl Crawly.Pipeline
+  def run(item, state, opts \\ []) do
+    job_tag =
+      case Map.get(state, :job_tag, nil) do
```

Comment:
I'd refactor this to:
```
state
|> Map.get(:job_tag, UUID.uuid1())
...
```

Reply:
I'd also use a with/do/else instead of a having multiple case/do when all you're doing is binding variables.

Reply:
good!